### PR TITLE
API-36070-split-method-error

### PR DESCRIPTION
**Pull request by stiehlrod**
## Summary

- Adds ampersand safety,
- Adds ternary in case max_completed_phase is nil.

## Related issue(s)

- [API-36070](https://jira.devops.va.gov/browse/API-36070)

## Testing done

- rspec
- rails counsel
- Postman

## What areas of the site does it impact?
	modified:   modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb [hunk 1]
```diff
@@ -262,20 +262,20 @@ def get_completed_phase_number_from_phase_details(details)
...
-              completed_phase_number = phase[:phase_type_change_ind].split('').first
-              if completed_phase_number < max_completed_phase
+            lc_status_array&.reverse&.map do |phase|
+              completed_phase_number = phase[:phase_type_change_ind]&.split('')&.first
+              if completed_phase_number < (max_completed_phase.nil? ? 0 : max_completed_phase)
                 phase_date["phase#{completed_phase_number}CompleteDate"] = date_present(phase[:phase_chngd_dt])
               end
             end
-          end.sort.reverse.to_h
+          end&.sort&.reverse.to_h
```
Comment (rockwellwindsor-va) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:278]: @stiehlrod  What are your thoughts on if this is worth adding a test for?

**Review by mchristiansonVA — DISMISSED** [review]
Tested claims show locally, debugging to set various aspects of the BGS claim details dto nil in `get_bgs_phase_completed_dates`; show endpoint handled gracefully and returned results as expected in Postman. LGTM!

File: modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb [hunk 2]
```diff
@@ -262,20 +262,26 @@ def get_completed_phase_number_from_phase_details(details)
...
           lc_status_array =
-            [data&.dig(:benefit_claim_details_dto, :bnft_claim_lc_status)].flatten.compact
-          return {} if lc_status_array.first.nil?
+            [data&.dig(:benefit_claim_details_dto, :bnft_claim_lc_status)].flatten&.compact
+          return {} if lc_status_array&.first&.nil?
 
-          max_completed_phase = lc_status_array.first[:phase_type_change_ind].split('').last
-          return {} if max_completed_phase.downcase.eql?('n')
+          max_completed_phase = lc_status_array&.first&.[](:phase_type_change_ind)&.split('')&.last
+          return {} if max_completed_phase&.downcase.eql?('n')
```
Comment (FonzMP) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:269]: we can emulate this with the example we received from VA.gov - TLDR below:
```
data = {:benefit_claim_details_dto=>{:bnft_claim_lc_status=>{:phase_type=>"Unknown"}}}
lc_status_array = [data&.dig(:benefit_claim_details_dto, :bnft_claim_lc_status)].flatten&.compact

lc_status_array
 => [{:phase_type=>"Unknown"}]

lc_status_array.first.nil?
 => false

max_completed_phase = lc_status_array.first[:phase_type_change_ind].split('').last
(irb):34:in `<main>': undefined method `split' for nil:NilClass (NoMethodError)
```

if we add :268 ( safe navigation ) I think we return out (`{}`) at :269, so add an `||` there if that's the expectation. Then we wouldn't need any of the change in :272 - :284

I'm not sure how we're just now seeing these `{:phase_type=>"Unknown"}` and maybe that's the better thing to understand?

**Review by FonzMP — APPROVED** [review]
:shipit:

**Pull request merged**